tests/integration/targets/mso_user/tasks/main.yml

```diff
@@ -274,10 +389,30 @@
     - nm_query_user.current.username == 'ansible_test'
     - cm_query_user == nm_query_user
 
+- name: Query our read-only user
+  mso_user:
+    <<: *user_query
+    user: ansible_test_read
+  register: nm_query_user2
+
+- name: Verify query_user2
+  assert:
+    that:
+    - nm_query_user2 is not changed
+    - nm_query_user2.current.roles[0].accessType == 'readOnly'
 
 # REMOVE USER
 - name: Remove user (check_mode)
-  mso_user: *user_absent
+  mso_user:
```

Comment:
You should add an anchor here and use it for the next 3 steps.
Like :
mso_user:  &user_absent